tests/v2/validation/public_api/crdgeneration/crd_generation.go

```diff
@@ -0,0 +1,137 @@
+package crdgeneration
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	v1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"github.com/rancher/rancher/tests/framework/extensions/kubeapi/customresourcedefinitions"
+	"github.com/rancher/rancher/tests/framework/extensions/kubectl"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	groupManagement   = "management.cattle.io"
+	groupCatalog      = "catalog.cattle.io"
+	groupUI           = "ui.cattle.io"
+	groupCluster      = "cluster.cattle.io"
+	groupProject      = "project.cattle.io"
+	groupRKE          = "rke.cattle.io"
+	groupProvisioning = "provisioning.cattle.io"
+	groupFleet        = "fleet.cattle.io"
+	roleTemplate      = "roletemplates.management.cattle.io" //Remove in 2.8.0
+	fleetlocal        = "fleet-local"
+	fleetdefault      = "fleet-default"
+)
+
+func mapCRD(crdList []string) map[string][]string {
+	groups := []string{groupManagement, groupCatalog, groupUI, groupCluster, groupProject, groupRKE, groupProvisioning, groupFleet}
+	crdMap := make(map[string][]string)
+
+	for _, group := range groups {
+		crdMap[group] = []string{}
+	}
+
+	crdMap["additional"] = []string{}
+
+	for _, crd := range crdList {
+		found := false
+		for _, group := range groups {
```

Comment:
I think this could be more efficient, but if these lists are generally short then it should be fine as is.

Reply:
These are generally short so keeping them as is.